app/connectors/TrustDataConnector.scala

```diff
@@ -40,15 +41,15 @@ class TrustDataConnector @Inject()(http: HttpClient, config: AppConfig) extends
       CORRELATION_ID -> correlationId
     )
 
-  def getTrustJson(identifier: Identifier)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[TrustDataResponse] = {
+  def getTrustJson(identifier: Identifier): Future[TrustDataResponse] = {
     lazy val url: String = s"${config.trustDataUrl}/trusts/obliged-entities/$identifier/${identifier.value}"
 
     val correlationId = UUID.randomUUID().toString
 
-    val headersWithoutOldAuth = hc.copy(authorization = None)
-    val hcExtra: HeaderCarrier = headersWithoutOldAuth.withExtraHeaders(trustDataHeaders(correlationId): _*)
+    implicit val hc: HeaderCarrier = HeaderCarrier(authorization = None, extraHeaders = trustDataHeaders(correlationId))
     logger.info(s"[Session ID: ${Session.id(hc)}] getTrustJson correlationId: $correlationId from call to url: $url")
-    http.GET[TrustDataResponse](url)(TrustDataResponse.httpReads(identifier), hcExtra, ec)
+
+    http.GET[TrustDataResponse](url)(TrustDataResponse.httpReads(identifier), implicitly[HeaderCarrier](hc), global)
```

Comment:
Out of interest, how does `implicitly[HeaderCarrier](hc)` differ from just passing in `hc`?

Reply:
I don't know. I just copied exactly what Trusts is doing. I had the same thought that it probably would do nothing, but I just want the call to work so thought it was best to copy it exactly

Reply:
Fair enough